libfaad/syntax.c

```diff
@@ -344,6 +344,12 @@ static void decode_sce_lfe(NeAACDecStruct *hDecoder,
        can become 2 when some form of Parametric Stereo coding is used
     */
 
+    if (hDecoder->frame && hDecoder->element_id[hDecoder->fr_ch_ele] != id_syn_ele) {
+        /* element inconsistency */
+        hInfo->error = 21;
+        return;
+    }
+
     /* save the syntax element id */
     hDecoder->element_id[hDecoder->fr_ch_ele] = id_syn_ele;
```

Comment:
I guess this line becomes irrelevant, then?

Reply:
No, I don't think so. In the case of the first frame, hDecoder->frame is 0 (it is incremented by libfaad/sbr_dec.c) and hDecoder->element_id[hDecoder->fr_ch_ele] is not set yet.

But we could very well put that in the "else". It would just make the diff bigger.

Reply:
Agree! Comments would not be bad, especially because the SBR and PCE related code are far from trivial. I see you have merged the PR, do you still want me to update anything?

Reply:
I merged the PR, because I found your new commit message far more descriptive than before. If you could push another commit with some explanatory code comments, that would still be nice, though!

Reply:
will do along with future fixes. Thanks!